scripts/zkai_attention_kv_fused_softmax_table_microprofile_gate.py

```
@@ -0,0 +1,920 @@
+    if buckets["opening_bucket_bytes"] != section_bytes["decommitments"] + section_bytes["fri_proof"]:
+        raise FusedSoftmaxTableMicroprofileGateError("opening bucket drift")
+    if row["lookup_relation_width"] is None:
+        if row["lookup_relation_width_status"] != RELATION_WIDTH_STATUS_INFERRED_MISSING:
+            raise FusedSoftmaxTableMicroprofileGateError("lookup relation width status drift")
+    else:
+        require_int(row["lookup_relation_width"], "lookup_relation_width")
+        if row["lookup_relation_width_status"] != RELATION_WIDTH_STATUS_EXPOSED:
+            raise FusedSoftmaxTableMicroprofileGateError("lookup relation width status drift")
+    component_rows = row["trace_rows_by_component"]
+    if not isinstance(component_rows, dict):
+        raise FusedSoftmaxTableMicroprofileGateError("trace component rows must be object")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)